feature/experimental/p4rt/ate_tests/lldp_packetin_test/lldp_packetin_test.go

```diff
@@ -0,0 +1,493 @@
+								t.Fatalf("Packet in PacketIn message is not matching wanted packet.")
+							}
+						}
+
+						metaData := packet.Pkt.GetMetadata()
+						for _, data := range metaData {
+							if data.GetMetadataId() == metadataIngressPort {
+								if string(data.GetValue()) != args.packetIO.GetIngressPort() {
+									t.Fatalf("Ingress Port Id is not matching expectation.")
+								}
+							}
+							if data.GetMetadataId() == metadataEgressPort {
+								found := false
+								for _, portData := range args.packetIO.GetEgressPort() {
+									if string(data.GetValue()) == portData {
+										found = true
+									}
+								}
+								if !found {
+									t.Fatalf("Egress Port Id is not matching expectation.")
+								}
+
+							}
+						}
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// sortPorts sorts the ports by the testbed port ID.
+func sortPorts(ports []*ondatra.Port) []*ondatra.Port {
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	return ports
+}
+
+// configInterfaceDUT configures the interface with the Addrs.
+func configInterfaceDUT(i *oc.Interface, a *attrs.Attributes) *oc.Interface {
+	i.Description = ygot.String(a.Desc)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	s4a := s4.GetOrCreateAddress(a.IPv4)
+	s4a.PrefixLength = ygot.Uint8(ipv4PrefixLen)
+
+	return i
+}
+
+// configureDUT configures port1 and port2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+
+	p1 := dut.Port(t, "port1")
+	i1 := &oc.Interface{Name: ygot.String(p1.Name())}
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(i1, &dutPort1))
+
+	p2 := dut.Port(t, "port2")
+	i2 := &oc.Interface{Name: ygot.String(p2.Name())}
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(i2, &dutPort2))
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	i1 := top.AddInterface(atePort1.Name).WithPort(p1)
+	i1.IPv4().
+		WithAddress(atePort1.IPv4CIDR()).
+		WithDefaultGateway(dutPort1.IPv4)
+
+	p2 := ate.Port(t, "port2")
+	i2 := top.AddInterface(atePort2.Name).WithPort(p2)
+	i2.IPv4().
+		WithAddress(atePort2.IPv4CIDR()).
+		WithDefaultGateway(dutPort2.IPv4)
+
+	return top
+}
+
+// configureDeviceID configures p4rt device-id on the DUT.
+func configureDeviceID(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
Check existing p4rt tests in featureprofiles/feature/experimental/p4rt for the latest changes in how to configure

Reply:
Done.